Sources/GRPCCore/Call/ClientRequest.swift

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2023, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/// A namespace for request message types used by clients.
+public enum ClientRequest {}
+
+extension ClientRequest {
+  /// A request created by the client for a single message.
+  ///
+  /// This is used for unary and server-streaming RPCs.
+  ///
+  /// See ``ClientRequest/Stream`` for streaming requests and ``ServerRequest/Single`` for the
+  /// servers representation of a single-message request.
+  ///
+  /// ## Creating ``Single`` requests
+  ///
+  /// ```swift
+  /// let request = ClientRequest.Single<String>(message: "Hello, gRPC!")
+  /// print(request.metadata)  // prints '[:]'
+  /// print(request.message)  // prints 'Hello, gRPC!'
+  /// ```
+  public struct Single<Message: Sendable>: Sendable {
+    /// Caller-specified metadata sent to the server at the start of the RPC.
+    ///
+    /// Both gRPC Swift and its transport layer may insert additional metadata. Keys prefixed with
+    /// "grpc-" are prohibited and may result in undefined behaviour. Transports may also insert
+    /// their own metadata, you should avoid using key names which may clash with transport specific
+    /// metadata. Note that transports may also impose limits in the amount of metadata which may
+    /// be sent.
+    public var metadata: Metadata
+
+    /// The message to send to the server.
```

Comment:
Extreme ultra nit: Above you say `metadata sent to the server` and here you say `message to send to the server`. Do we want to align this.